httplug/exceptions.rst

```diff
@@ -0,0 +1,20 @@
+Exceptions
+==========
+
+HTTPlug defines one Exception interface: ``Http\Client\Exception``. All HTTP
+client exceptions must implement this interface.
+
+``HttpClient::sendRequest()`` can throw one of the following exceptions.
+
+================================== ====================== ===================
+Exception                          Thrown when            Methods available
+================================== ====================== ===================
+TransferException                  ???
+└ RequestException                 the request is invalid ``getRequest()``
+ |nbsp| |nbsp| └ NetworkException  no response received
+                                   due to network issues  ``getRequest()``
+ |nbsp| |nbsp| └ HttpException     error response         ``getRequest()``
+                                                          ``getResponse()``
+================================== ====================== ===================
```

Comment:
for example here, when we don't know exactly what is happening: https://github.com/php-http/guzzle6-adapter/blob/master/src/Promise.php#L138